### ISSUE TITLE
Dialog: Screen readers can access background content when Dialog is open

### ISSUE THREAD

**Issue by Faiz-exe** [primefaces/primereact#7974]
### Describe the bug

When running accessibility tests using tools like axe-core, the Dialog component reports the following issue:
**"Screen readers can read parent page content outside the modal"**
more info on this issue can be read here : https://docs.deque.com/issue-help/1.0.0/en/reading-order-browse-outside-modal

This suggests that screen readers can still access the background content even when the modal is open. Ideally, background content should be hidden from assistive technologies when a modal is active.

I can see that the focus is trapped in the dialog , however I still get this issue

### System Information

```Shell
Primereact Version: 10.9.1
Browser: Chrome
Operating System: Windows
Browser: [e.g. Chrome, Firefox]
Accessibility tool used: Axe, axe-core, axe-cypress
```

### Steps to reproduce the behavior

1. Open a page with a PrimeReact Dialog component.
2. Trigger the dialog to open.
3. Run an accessibility scan using axe devtools ( IGT mode , select Modal )
4. Observe the reported violation.

### Expected behavior

The content outside the dialog should be hidden from assistive technologies

**Comment by melloware** [comment]
From reading that its not quite as easy as it seems it says that every other sibling in the `body` with the dialog has to get `aria-hidden="true"`.  And then have aria-hidden removed when the dialog is closed.
